src/core/basicfb.cpp

```diff
@@ -195,12 +202,14 @@ void CBasicFB::traceInstanceData() {
     internals_c_str[i] = valueString.c_str();
   }
 
-  for(TPortId i = 0; i < internalFbs.size(); ++i) {
-    CFunctionBlock *value = getInternalFB(i);
+  TPortId i = 0;
+  for(TFBContainerList::iterator it(getChildren().begin()); it != getChildren().end(); ++it){
```

Comment:
I would suggest either iterating from `0` to `getChildren().size()` or using a [range-for](https://en.cppreference.com/w/cpp/language/range-for) loop.